src/pg_to_evalscript/javascript_datacube/DataCube.js

```diff
@@ -58,7 +58,32 @@ class DataCube {
       }
 
     removeDimension(dimension) {
-        this.dimensions = this.dimensions.filter(d => d.name !== dimension)
+        const idx = this.dimensions.findIndex(d => d.name === dimension);
+        this.dimensions = this.dimensions.filter(d => d.name !== dimension);
+        this.removeDimensionFromData(this.data, 0, idx === 0 ? idx : idx - 1);
+    }
+
+    removeDimensionFromData(data, currLevel, level) {
+        if (currLevel === level) {
+            if (currLevel === 0) {
+                const tmpData = data[0]
+                for (let i = 0; i < tmpData.length; i++) {
+                    data[i] = tmpData[i]
+                }
+                return;
+            }
+
+            for (let j = 0; j < data.length; j++) {
+                const tmpData = data[j]
+                for (let i = 0; i < tmpData.length; i++) {
+                    data[j] = tmpData[i]
```

Comment:
For simpler understanding, everything inside the outer `for` could be rewritten as:
```js
data[j] = lastElement(data[j]) // or data[j] = data[j][data[j].length-1] using JS syntax
```
`data` needs to be an array of length 1 if we want to drop the dimension, other wise that doesn't make any sense. So it could be just `data[j] = data[j][0]`.

However, this is actually dropping the dimension a level lower than we are currently at. That's why `idx === 0 ? idx : idx - 1` is done. This of course introduces bugs.

For example, say our data is
```
[[1],[2],[3]]
```
And we want to drop the second dimension (index 1). We would expect `[1,2,3]`. However:
* calling `removeDimensionFromData(data,0,1)` doesn't do anything, as we enter this second loop and get e.g `[1]` as `data` and just sets `data[0] = 1`
* calling `removeDimensionFromData(data,0,0)` as would come from `removeDimension` enters `if (currLevel === 0)`. That one just sets the first element to `1`.

---------------------------------

Anyway, this function could also be implemented using existing methods defined on the `DataCube`, namely `_iter`.

e.g I believe this should work:
```js
removeDimension(dimension) {
    const idx = this.dimensions.findIndex(d => d.name === dimension);
    this.dimensions = this.dimensions.filter(d => d.name !== dimension);
    this.data = this._iter(this.data, v => v, coords => coords.length === idx)
}
```
It ultimately doesn't matter if it's implemented separately or not, because the implementation of `DataCube` based on arrays-of-arrays will be changed soon as a part of performance optimisations.

Reply:
Ah I see the bug now yes..

This solution with the `_iter` function works like a charm I think.